### ISSUE TITLE
Make it possible to restrict processes to certain users/machines

### ISSUE THREAD

**Issue by hbpe-aarhus** [itk-dev-rpa/OpenOrchestrator#82]
Currently, if multiple schedulers are connected to a single orchestrator database, it is not possible to control which user/machine combination executes a specific process.

We would like to be able to dedicate certain combinations of user/machine to specific processes. Both to reduce the number of orchestrator databases needed, but also to dedicate resources to time-critical processes.

Restriction could either be at process-level or at trigger-level.

A possible solution could be to let each scheduler have its own whitelist of processes that it is allowed to execute. The whitelist could be an extra option that needs to be enabled such that current setups are unaffected.
Alternatively, one could make it possible to configure from the orchestrator GUI.

**Comment by ghbm-itk** [comment]
This should be possible to implement.
It would probably be best if it's controlled in the Orchestrator application.
It could be given as a list of pc names on the trigger where an empty list means that it can run wherever.

**Comment by hbpe-aarhus** [comment]
Yes, I agree that controlling it centrally from the Orchestrator application would be preferable.

Am I correct in assuming that it's not feasible to run several service accounts on the same machine? If so, the machine name would be sufficient, as you suggest.

**Comment by hbpe-aarhus** [comment]
We recently discussed this topic.

The idea is to have a Scheduler dedicated to a specific "area of business" (forretningsområde), where processes are time-critical and must run at specific timestamps throughout a standard working day.

Simply restricting a Trigger isn’t sufficient, is it? A Scheduler could still be started by another long-running Trigger, potentially blocking the execution of other Triggers.

An example to illustrate what I'm thinking:

* Scheduler A can run any Trigger
* Scheduler B can run only Trigger I and II

However, if we restrict Triggers I and II to Scheduler B, Scheduler B could still be started by another Trigger, leading to potential blocking of Scheduler B for an extended period.

**Comment by ghbm-itk** [comment]
The problem would be solved by setting a restriction on all your triggers.

**Comment by hbpe-aarhus** [comment]
While technically correct, it would also be very tedious and error-prone (it's easy to forget when you're setting up a new Trigger).

**Comment by ghbm-itk** [comment]
I agree. So it should be possible to tell the scheduler, that it should only touch its own triggers.
In your example, should Scheduler A be able to run Trigger I and II?

**Comment by hbpe-aarhus** [comment]
Yes, precisely. Our wish is that a Scheduler should only run its own triggers :)

Correct, Scheduler A should not run Trigger I and II.
